data-prepper-plugins/mutate-string-processors/README.md

```diff
@@ -0,0 +1,148 @@
+# Mutate String Processors
+The following is a list of processors to mutate a string.
+
+## SubstituteStringProcessor
+A processor that matches a field value against a regular expression and replaces all matches with a replacement string.
+
+### Basic Usage
```

Comment:
@sshivanii, this section only has the creation of a pipeline file. It doesn't provide details for how to send data and data requirements for this configuration to work and may be a little confusing to our users. To avoid having to repeat steps for creating a file, changing the data structure, etc. for each processor for basic usage, I would recommend we focus on just the processor configuration options and a simple example.  

### Example
The following `substitute_string` processor example will replace all ':' with '_' for the `message` key.

```yaml
  processor:
    - substitute_string:
        entries:
          - source: "message"
            from: ":"
            to: "_"
```

Reply:
Yes it does seem redundant. I like your suggestion and will add examples instead of basic usage.